src/main/java/eu/europeana/api/recommend/util/RecommendServiceUtils.java

```diff
@@ -0,0 +1,170 @@
+package eu.europeana.api.recommend.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+
+import java.util.*;
+
+public class RecommendServiceUtils {
+
+    private static final Logger LOG = LogManager.getLogger(RecommendServiceUtils.class);
+
+    private static final String SOLR_ID_FIELD = "europeana_id";
+    private static final String ITEMS_FIELD = "items";
+    private static final String PREF_LABEL_FIELD = "prefLabel";
+    private static final String ALT_LABEL_FIELD = "altLabel";
+
+
+    /**
+     * Constructs a Search API query in the form
+     * <pre>query=europeana_id:("/x1/y1" OR "/x2/y2 OR "/x3/y3")&pageSize=3&profile=minimal&wskey=[wskey]</pre>
+     */
+    public String generateSearchQuery(String[] recordIds, int maxResults, String wskey) {
+        StringBuilder s = new StringBuilder(50).append("?query=");
+        for  (int i = 0; i < maxResults && i < recordIds.length; i++) {
+            if (i > 0) {
+                s.append(" OR ");
+            }
+            s.append(SOLR_ID_FIELD).append(":\"").append(recordIds[i]).append('"');
+        }
+        s.append("&rows=").append(recordIds.length)
+                .append("&profile=minimal")
+                .append("&wskey=").append(wskey);
+        return s.toString();
+    }
+
+    /**
+     * Generates Entity Api search query to fetch all skos:preflabel and skos:altlabel
+     * for the given entity uri.
+     *
+     * example : ?query=entity_uri:%22<entityUri>%22&fl=skos_prefLabel.*,skos_altLabel.*&wskey=
+     * @param entityUri
+     * @param wskey
+     * @return
+     */
+    public String entityApiSearchQuery(String entityUri, String wskey) {
+        StringBuilder query = new StringBuilder("?query=entity_uri:");
+        query.append("\"" + entityUri + "\"");
```

Comment:
If we are using a StringBuilder to append, best to use it for everything and avoid also using concatenation.

Reply:
done